resources/service.rb

```diff
@@ -0,0 +1,150 @@
+# Copyright:: Copyright (c) 2017 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+resource_name :hab_service
+provides :hab_service
+
+property :service_name, String, name_property: true
+property :loaded, [true, false], default: false, desired_state: true
+property :running, [true, false], default: false, desired_state: true
+
+# hab sup options which get included based on the action of the resource
+property :permanent_peer, [true, false], default: false
+property :listen_gossip, String
+property :listen_http, String
+property :org, String, default: 'default'
+property :peer, String
+property :ring, String
+property :strategy, String
+property :topology, String
+property :depot_url, String
+property :bind, String
+property :service_group, String
+property :config_from, String
+property :override_name, String, default: 'default'
+
+load_current_value do
+  require 'json'
```

Comment:
should just use ffi-yajl or chef/json_compat instead of json gem.

actually you should probably use Chef::HTTP::SimpleJSON